hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -206,7 +207,11 @@ private static GenericRecord rewrite(GenericRecord record, LinkedHashSet<Field>
     GenericRecord newRecord = new GenericData.Record(newSchema);
     for (Schema.Field f : fieldsToWrite) {
       if (record.get(f.name()) == null) {
-        newRecord.put(f.name(), f.defaultVal());
+        if (f.defaultVal() instanceof Null) {
+          newRecord.put(f.name(), null);
```

Comment:
Why do you need this? If f.defaultVal() is null, it will be automatically populated like that.

Reply:
just run the test without it. You will get an exception. 
if schema has "default": null it breaks, which should not be the case.

Reply:
Which test you are talking about?

Reply:
the one that was modified in this PR: TestHoodieAvroUtils.java. I updated test avro schema.

Reply:
But these tests are passing on the master branch without the patch that you did. :)

Reply:
Ok so when using f.defaultVal(), NullNode corresponds to JsonProperties.Null class and hence the test case fails. Good catch :)